### DISCUSSION TITLE
Possibility to compare only a single (or list of) databases

### DISCUSSION THREAD

**Discussion by dlaplante75** [littleK0i/SnowDDL#6] [Ideas]
Hi! I think this project is great!

I would be great to have the possibility to deploy only a single DB (or list of DBs). For example, my snowflake account has about 10 databases, my snowddl project has only 1. 

Deploying the project (safe deploy only) always outputs a whole list of suggested DDL to drop everything else... thankfully, they are just suggestions. But what if I DO want to drop from the database, things that are not in my project. If I enable unsafe deployment, it will drop EVERYTHING. I would like for the deployment to only consider specific databases.

**Comment by littleK0i** [comment]
The best method to achieve this goal right now is to run SnowDDL with user having `OWNERSHIP` only on databases which are supposed to be managed by SnowDDL.

SnowDDL will not touch databases and schemas which are not directly "owned" by the role of SnowDDL user.

The specific places in code with this check:

- databases: https://github.com/littleK0i/SnowDDL/blob/master/snowddl/schema_cache.py#L25-L27
- schemas: https://github.com/littleK0i/SnowDDL/blob/master/snowddl/schema_cache.py#L53-L55

I am considering the "singledb" feature, but it will also skip management of roles & grants.

Having an option to include / exclude certain databases by name seems great at first, but ... at some point someone will make a mistake or typo in command and drop everything. In my opinion, the best way to "protect" other objects is by not granting SnowDDL the ability to destroy them in the first place.

I hope it makes sense :)

**Reply by dlaplante75** [reply]
It does! Thanks!

**Comment by littleK0i** [comment]
@dlaplante75 , please note

The SingleDB mode with documentation is now available: https://docs.snowddl.com/single-db/overview

Please update to version `0.6.0` or higher to see it in action.

Enjoy!

**Reply by dlaplante75** [reply]
Super cool ! Thank you so much ! can't wait to try it out !

**Reply by minghungcho** [reply]
While testing SingleDB mode, I noticed --apply-unsafe still only prints out "Suggested DDL", Any idea how I can force SnowDDL to apply CREATE OR REPLACE TABLE changes?

I have account, user, role, ... stored in environment. so the following is the command I used.

snowddl-singledb -c test --show-sql --exclude-object-types SCHEMA --target-db test_db --apply-unsafe apply

**Reply by littleK0i** [reply]
@minghungcho , it is possible to apply `CREATE OR REPLACE TABLE` using flag `--apply-replace-table` in addition to `--apply-unsafe`.

The main reason for a separate flag is that such change is not only "unsafe", but it also requires an active warehouse and full table overwrite. It may take a long time and use a lot of credits, so I usually suggest to review such changes and apply manually.

Full list of `--apply-*` flags is available in documentation: https://docs.snowddl.com/basic/cli
